src/go/k8s/controllers/redpanda/cluster_controller_configuration_drift.go

```diff
@@ -0,0 +1,203 @@
+	defer log.V(debugLogLevel).Info(fmt.Sprintf("Finished configuration drift reconcile loop for %v", req.NamespacedName))
+
+	var redpandaCluster redpandav1alpha1.Cluster
+	if err := r.Get(ctx, req.NamespacedName, &redpandaCluster); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("unable to retrieve Cluster resource: %w", err)
+	}
+
+	if !isRedpandaClusterManaged(log, &redpandaCluster) {
+		return ctrl.Result{RequeueAfter: r.getDriftCheckPeriod()}, nil
+	}
+
+	condition := redpandaCluster.Status.GetCondition(redpandav1alpha1.ClusterConfiguredConditionType)
+	if condition == nil || condition.Status != corev1.ConditionTrue {
+		// configuration drift already signaled
+		return ctrl.Result{RequeueAfter: r.getDriftCheckPeriod()}, nil
+	}
+
+	// wait at least a driftCheckPeriod before checking drifts
+	now := time.Now()
+	driftCheckPeriod := r.getDriftCheckPeriod()
+	if condition.LastTransitionTime.Time.Add(driftCheckPeriod).After(now) {
+		period := condition.LastTransitionTime.Time.Add(driftCheckPeriod).Sub(now)
+		return ctrl.Result{RequeueAfter: period}, nil
+	}
+
+	redpandaPorts := networking.NewRedpandaPorts(&redpandaCluster)
+	headlessPorts := collectHeadlessPorts(redpandaPorts)
+	clusterPorts := collectClusterPorts(redpandaPorts, &redpandaCluster)
+
+	headlessSvc := resources.NewHeadlessService(r.Client, &redpandaCluster, r.Scheme, headlessPorts, log)
+	clusterSvc := resources.NewClusterService(r.Client, &redpandaCluster, r.Scheme, clusterPorts, log)
+
+	var proxySu *resources.SuperUsersResource
+	var proxySuKey types.NamespacedName
+	if redpandaCluster.Spec.EnableSASL && redpandaCluster.PandaproxyAPIInternal() != nil {
+		proxySu = resources.NewSuperUsers(r.Client, &redpandaCluster, r.Scheme, resources.ScramPandaproxyUsername, resources.PandaProxySuffix, log)
+		proxySuKey = proxySu.Key()
+	}
+	var schemaRegistrySu *resources.SuperUsersResource
+	var schemaRegistrySuKey types.NamespacedName
+	if redpandaCluster.Spec.EnableSASL && redpandaCluster.Spec.Configuration.SchemaRegistry != nil {
+		schemaRegistrySu = resources.NewSuperUsers(r.Client, &redpandaCluster, r.Scheme, resources.ScramSchemaRegistryUsername, resources.SchemaRegistrySuffix, log)
+		schemaRegistrySuKey = schemaRegistrySu.Key()
+	}
+	pki := certmanager.NewPki(r.Client, &redpandaCluster, headlessSvc.HeadlessServiceFQDN(r.clusterDomain), clusterSvc.ServiceFQDN(r.clusterDomain), r.Scheme, log)
+	configMapResource := resources.NewConfigMap(r.Client, &redpandaCluster, r.Scheme, headlessSvc.HeadlessServiceFQDN(r.clusterDomain), proxySuKey, schemaRegistrySuKey, log)
+
+	lastAppliedConfig, cmExists, err := configMapResource.GetLastAppliedConfigurationFromCluster(ctx)
+	if err != nil {
+		log.Error(err, "could not get last applied configuration to check drifts")
+		return ctrl.Result{RequeueAfter: r.getDriftCheckPeriod()}, nil
```

Comment:
So, returning an error makes the first arg to be ignored by controller-runtime (https://github.com/kubernetes-sigs/controller-runtime/blob/b9219528d95974cb4f5b06f86c9b1c9b7d3045a5/pkg/internal/controller/controller.go#L322-L343). I wanted to have a more consistent execution, to avoid polluting logs and not incur in exponential backoff delay..

But, thinking again, since errors here should not be common, I think it should not be a problem to just return them, also to have them counted in the metrics..